src/devices/x10_rf.c

```diff
@@ -17,7 +17,7 @@ static int X10_RF_callback(bitbuffer_t *bitbuffer) {
 	// Validate package
 	if ((bitbuffer->bits_per_row[1] == 32)		// Dont waste time on a short package
 	// && (bb[1][0] == (uint8_t)(~bb[1][1]))		// Check integrity - apparently some chips may use both bytes..
-	 && (bb[1][2] == (uint8_t)(~bb[1][3]))		// Check integrity
+	 && (bb[1][2] == (255 - bb[1][3]))		// Check integrity
 	)
```

Comment:
I had tried different casts, but it did not like any of them.
A separate variable was the only way i.e.
uint8_t x = ~bb[1][3];
bb[1][2] == x
Then I though well the ~ of a byte is the same as 255 - byte and that ended up not generating a compiler warning.